src/Microsoft.VisualStudio.ProjectSystem.Managed.VS.UnitTests/ProjectSystem/VS/TempPE/DesignTimeInputsBuildManagerBridgeTests.cs

```diff
@@ -0,0 +1,142 @@
+            Assert.Empty(_buildManager.DeletedItems);
+        }
+
+        [Fact]
+        public async Task RemovedFile_FiresTempPEDeleted()
+        {
+            await _bridge.TestProcessAsync(new DesignTimeInputsDelta(
+                ImmutableHashSet.CreateRange(new string[] { "Resources1.Designer.cs" }),
+                ImmutableHashSet<string>.Empty,
+                Array.Empty<DesignTimeInputFileChange>(),
+                ""));
+
+            await _bridge.TestProcessAsync(new DesignTimeInputsDelta(
+               ImmutableHashSet<string>.Empty,
+               ImmutableHashSet<string>.Empty,
+               Array.Empty<DesignTimeInputFileChange>(),
+               ""));
+
+            // One file should have been added
+            Assert.Empty(_buildManager.DirtyItems);
+            Assert.Single(_buildManager.DeletedItems);
+            Assert.Equal("Resources1.Designer.cs", _buildManager.DeletedItems.First());
+        }
+
+        [Fact]
+        public async Task GetDesignTimeInputXmlAsync_HasCorrectArguments()
+        {
+            await _bridge.TestProcessAsync(new DesignTimeInputsDelta(
+                ImmutableHashSet.CreateRange(new string[] { "Resources1.Designer.cs" }),
+                ImmutableHashSet<string>.Empty,
+                new DesignTimeInputFileChange[] { new DesignTimeInputFileChange("Resources1.Designer.cs", false) },
+                "C:\\TempPE"));
+
+            await _bridge.GetDesignTimeInputXmlAsync("Resources1.Designer.cs");
+
+            Assert.Equal("Resources1.Designer.cs", _lastCompiledFile);
+            Assert.Equal("C:\\TempPE", _lastOutputPath);
+            Assert.Equal(ImmutableHashSet<string>.Empty, _lastSharedInputs);
+        }
+
+        public DesignTimeInputsBuildManagerBridgeTests()
+        {
+            var threadingService = IProjectThreadingServiceFactory.Create();
+
+            var changeTracker = Mock.Of<IDesignTimeInputsChangeTracker>();
+
+            var compilerMock = new Mock<IDesignTimeInputsCompiler>();
+            compilerMock.Setup(c => c.GetDesignTimeInputXmlAsync(It.IsAny<string>(), It.IsAny<string>(), It.IsAny<ImmutableHashSet<string>>()))
+                .Callback<string, string, ImmutableHashSet<string>>((file, outputPath, sharedInputs) =>
+                {
+                    _lastCompiledFile = file;
+                    _lastOutputPath = outputPath;
+                    _lastSharedInputs = sharedInputs;
+                });
+
+
+            _buildManager = new TestBuildManager();
+
+            _bridge = new TestDesignTimeInputsBuildManagerBridge(threadingService, changeTracker, compilerMock.Object, _buildManager);
+            _bridge.SkipInitialization = true;
+        }
+
+        public void Dispose()
+        {
+            ((IDisposable)_bridge).Dispose();
+        }
+
+        internal class TestDesignTimeInputsBuildManagerBridge : DesignTimeInputsBuildManagerBridge
```

Comment:
Does this class only exist to provide a helper method for ApplyAsync?

Reply:
Yes, it only exists because ApplyAsync is protected